src/core/connection.c

```diff
@@ -5637,6 +5637,9 @@ QuicConnRecvDatagrams(
 
         if (!IsDeferred) {
             Connection->Stats.Recv.TotalBytes += Packet->BufferLength;
+            if (Connection->Stats.Handshake.HandshakeHopLimitTTL == 0) {
+                Connection->Stats.Handshake.HandshakeHopLimitTTL = Packet->HopLimitTTL;
```

Comment:
You are using zero as a sentinel value. So, a valid TTL value should never be zero? (it would make sense for zero TTL packets to be dropped anyways but I am just double checking here).

Reply:
Good points about whether using 0 as a sentinel value is correct. 
I think we can do it this way, but @nibanks can chime in if not.

In terms of where to set this value, setting it here is the simplest. It all depends on the perf impact.

Reply:
If you move it into the handshake code then you'll need to pass along packet info. I don't care much either way.

Reply:
No strong opinion. So feel free to ignore.